### DISCUSSION TITLE
--parameter-overrides using host Environment variables

### DISCUSSION THREAD

**Discussion by royassis** [aws/aws-sam-cli#5063] [Ideas]
Hey

Right now i'm using **samconfig.toml** to pass Parameters into my template.yaml.
The file is under version control and is uploaded to a remote repo. 

There is a secret which is not in **samconfig.toml** and I explicitly pass it via the command line, i.e. 
``` 
saws build --parameter-overrides mysecret=1234
```

It would would have been nice if it could be passed implicitly direct from my host environment parameters. 

Best,
Roy

**Comment by davidjb** [comment]
At the moment, I'm managing this with a Makefile so the env variables gets passed though automatically for the various `--parameter-overrides` I need to set. There's quite a few of them so writing them out by hand each time is impossible.

Ideally, SAM would support some manner of defining/loading Parameters from the host environment, allowing sensitive data to be used by SAM but not stored directly in `samconfig.[toml|yml]` or the template.
